diesel/src/sqlite/connection/raw.rs

```diff
@@ -388,3 +443,11 @@ where
     let ptr = data as *mut F;
     unsafe { Box::from_raw(ptr) };
 }
+
+extern "C" fn destroy_boxed_collation_fn<F>(data: *mut libc::c_void)
```

Comment:
Is there any reason why we can't reuse the `destroy_boxed_fn` function here. Technically this function does not need to know anything about `F`, therefore we can just remove the constraints there and use it everywhere.